tools/mtouch/Application.cs

```diff
@@ -1416,14 +1416,32 @@ void GenerateMSymManifest (Target target, string target_directory)
 			Driver.WriteIfDifferent (manifestPath, root.ToString ());
 		}
 
-		void CopyAssemblyData (Assembly asm, string targetDir)
+		void CopyAotData (string src, string dest)
 		{
-			var mvid = asm.AssemblyDefinition.MainModule.Mvid.ToString ().ToUpperInvariant ();
-			var assemblyDir = Path.Combine (targetDir, mvid);
-			if (!Directory.Exists (assemblyDir))
-				Directory.CreateDirectory (assemblyDir);
-			asm.CopyToDirectory (assemblyDir, reload: false, only_copy: true);
-			asm.CopyMSymToDirectory (targetDir);
+			if (string.IsNullOrEmpty (src) || string.IsNullOrEmpty (dest)) {
+				ErrorHelper.Warning (95, $"Aot files could not be copied to the destination directory {dest}"); 
+				return;
+			}
+				
+			var dir = new DirectoryInfo(src);
+			var dirs = dir.GetDirectories();
+
+			if (!dir.Exists) 
```

Comment:
You're calling `dir.GetDirectories()` before `dir.Exists`. Does that work if the directory doesn't exist?